common/src/main/java/org/sitmun/authorization/service/ProxyConfigurationService.java

```
@@ -0,0 +1,133 @@
+import org.sitmun.domain.service.parameter.ServiceParameter;
+import org.sitmun.domain.task.Task;
+import org.sitmun.domain.task.TaskRepository;
+import org.springframework.beans.factory.annotation.Value;
+
+@org.springframework.stereotype.Service
+public class ProxyConfigurationService {
+	
+	private final String SQL_COMMAND_KEY = "command";
+	
+	private final String CONNECTION_TYPE_KEY = "SQL";
+	
+	private final String VARY_KEY = "VARY";
+
+	private final CartographyRepository cartographyRepository;
+	
+	private final TaskRepository taskRepository;
+	
+	@Value("${sitmun.proxy.config-response-validity-in-seconds}")
+	private int responseValidityTime;
+	
+	public ProxyConfigurationService (CartographyRepository cartographyRepository,
+			TaskRepository taskRepository) {
+		this.cartographyRepository = cartographyRepository;
+		this.taskRepository = taskRepository;
+	}
+	
+	public boolean validateUserAccess(ConfigProxyRequest configProxyRequest, String userName) {
+		boolean result = true;
+		
+		return true;
+	}
+	
+	public ConfigProxyDto getConfiguration(ConfigProxyRequest configProxyRequest, String username) {
+		ConfigProxyDto result = null;
+		if(CONNECTION_TYPE_KEY.equalsIgnoreCase(configProxyRequest.getType())) {
+			Optional<Task> task = taskRepository.findById(configProxyRequest.getTypeId());
+			if (task.isPresent()) {
+				Task effectiveTask = task.get();
+				DatabaseConnection databaseConnection = effectiveTask.getConnection();
+				String sql = getSqlByTask(effectiveTask);
+				result = getDatasourceConfiguration(databaseConnection, sql);
+			}
+		}else {
+			Optional<Cartography> cartography = cartographyRepository.findById(configProxyRequest.getTypeId());
+			if(cartography.isPresent()) {
+				Cartography effectiveCartography = cartography.get();
+				Service service = effectiveCartography.getService();
+				result = getOgcWmsConfiguration(service, configProxyRequest.getParameters(), configProxyRequest.getMethod());
+			}
+		}
+		return result;
```

What happens when `result` is `null`? The request is probably wrong. Fail fast, throw an exception that returns a 401.